http/cors.go

```diff
@@ -0,0 +1,32 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/hashicorp/vault/logical"
+	"github.com/hashicorp/vault/vault"
+)
+
+func wrapCORSHandler(h http.Handler, core *vault.Core) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		corsConf := core.CORSConfig()
```

Comment:
> OPTIONS (or preflights) requests are made by the browser when a request is made with a method other than GET, HEAD, or POST and the content type is text/plain, application/x-www-form-urlencoded or multipart/form-data

All API calls in Vault are application/json. Doesn't this mean that CORS is totally unnecessary for Vault?